enumerable_methods.rb

```diff
@@ -0,0 +1,111 @@
+# This is the enumerable methods project done by Mohammad and Kingsley
+module Enumerable
+  def my_each
+    i = 0
+    object = self
+    while i < object.length
+      yield object[i]
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    object = self
+    while i < object.length
+      yield(object[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_arr = []
+    object = self
+    object.my_each do |x|
+      (new_arr << x) if yield(x) == true
```

Comment:
This conditional could be simplified by having it simply as `(new_arr << x) if yield(x)`